### PR TITLE
Add project parameter to additional resources

### LINKED ISSUE #7583: Certain CloudStack resources not added to tfstate
### Terraform Version
`0.6.16`

### Affected Resource(s)
Of the many resources in my configuration, the only two I've encountered with this issue are:
- `cloudstack_static_nat`
- `cloudstack_loadbalancer_rule`

### Terraform Configuration Files
```hcl
provider "cloudstack" {
  api_url           = "${var.api_url}"
  api_key           = "${var.api_key}"
  secret_key        = "${var.secret_key}"
}

resource "cloudstack_template" "template" {
  name              = "tf-template"
  format            = "VHD"
  hypervisor        = "XenServer"
  os_type           = "${var.os_type_id}"
  url               = "http://url/of/template"
  zone              = "-1"
  project           ="${var.project}"
}

resource "cloudstack_vpc" "main_vpc" {
  name              = "tf-default"
  cidr              = "${var.vpc_cidr}"
  vpc_offering      = "${var.vpc_offering}"
  zone              = "${var.zone}"
  project           = "${var.project}"
}

resource "cloudstack_network" "tier" {
  name              = "tf-tier"
  cidr              = "${cidrsubnet("${var.vpc_cidr}", 2, 1)}"
  network_offering  = "${var.network_offering}"
  vpc_id            = "${cloudstack_vpc.main_vpc.id}"
  zone              = "${var.zone}"
  acl_id            = "9ba3ec65-2e1d-11e4-8e05-42a29a39fc92"
  project           = "${var.project}"
}

resource "cloudstack_instance" "instance" {
  name              ="tf-instance"
  zone              ="${var.zone}"
  template          ="${cloudstack_template.template.id}"
  service_offering  ="${var.service_offering}"
  network_id        ="${cloudstack_network.tier.id}"
  project           = "${var.project}"
  keypair           = "${cloudstack_ssh_keypair.sshkey.id}"
  expunge           =false
}

resource "cloudstack_ipaddress" "static_nat_ip" {
  vpc_id            = "${cloudstack_vpc.main_vpc.id}"
  project           = "${var.project}"
}

resource "cloudstack_static_nat" "static_nat" {
  ip_address_id         = "${cloudstack_ipaddress.static_nat_ip.id}"
  virtual_machine_id    = "${cloudstack_instance.instance.id}"
  network_id            = "${cloudstack_network.tier.id}"
}

resource "cloudstack_ipaddress" "lb_ip" {
    vpc_id          = "${cloudstack_vpc.main_vpc.id}"
    project         = "${var.project}"
}

resource "cloudstack_loadbalancer_rule" "lbr" {
  name              = "tf-lbr"
  description       = "App load balancer rule"
  ip_address_id     = "${cloudstack_ipaddress.lb_ip.id}"
  algorithm         = "roundrobin"
  network_id        = "${cloudstack_network.tier.id}"
  private_port      = "33"
  public_port       = "33"
  member_ids        = ["${cloudstack_instance.instance.id}"]
}

resource "cloudstack_ssh_keypair" "sshkey" {
  name              = "tf-ssh-key"
  public_key        = "${file("~/.ssh/id_rsa.pub")}"
  project           = "${var.project}"
}
```

### Debug Output
1. Run `terraform apply` to spin up all the infrastructure
2. Run `terraform apply` again.
The debug output [here](https://gist.github.com/keith471/74f036f1d48f486f6d6b62d5ff01786b) was produced by the **second** step.

### Expected Behavior
Should not have attempted to create the load balancer rule nor the static NAT as they already exist.

### Actual Behavior
Attempts to create them, producing the following error output:
```
2 error(s) occurred:

* cloudstack_static_nat.static_nat: Error enabling static NAT: CloudStack API error 431 (CSExceptionErrorCode: 4350): Failed to enable static nat on the  ip 192.155.69.46 with Id 4102679d-f75d-4e9d-a885-d26f7912f7fd as the vm i-2649-12835-VM with Id cc5fb8b6-3d9c-4917-91b6-9b25733b7520 is already associated with another public ip 192.155.69.46 with id 4102679d-f75d-4e9d-a885-d26f7912f7fd
* cloudstack_loadbalancer_rule.lbr: CloudStack API error 537 (CSExceptionErrorCode: 9999): The range specified, 33-33, conflicts with rule 12294 which has 33-33
```

### Steps to Reproduce
1. `terraform apply`
2. `terraform plan` - you will see that the load balancer rule and static NAT are staged for creation
3. `terraform apply`

### References
It could be completely unrelated, but a similar-sounding past problem is [#2584](https://github.com/hashicorp/terraform/issues/2584).

### PR DESCRIPTION
**Pull request by svanharmelen**
Fixes #7583

### CODE REVIEW

File: builtin/providers/cloudstack/resource_cloudstack_loadbalancer_rule.go [hunk 1]
```diff
@@ -138,7 +145,10 @@ func resourceCloudStackLoadBalancerRuleRead(d *schema.ResourceData, meta interfa
 	cs := meta.(*cloudstack.CloudStackClient)
 
 	// Get the load balancer details
-	lb, count, err := cs.LoadBalancer.GetLoadBalancerRuleByID(d.Id())
+	lb, count, err := cs.LoadBalancer.GetLoadBalancerRuleByID(
+		d.Id(),
+		cloudstack.WithProject(d.Get("project").(string)),
```
Comment (stack72) [inline builtin/providers/cloudstack/resource_cloudstack_loadbalancer_rule.go]: What happens if this is a nil value?

Comment (svanharmelen) [inline builtin/providers/cloudstack/resource_cloudstack_loadbalancer_rule.go]: That will never happen... `d.Get()` will always give you a zero value if you didn't set it in your config.

**Comment by stack72** [comment]
After a short discussion with @svanharmelen on slack, passing an empty Project to `cloud.WithProject` is perfectly valid :)

therefore, this LGTM!

**Comment by svanharmelen** [comment]
Thanks for the quick view!

**Pull request merged**